### PR TITLE
refactor: Use dexie-observable to handle message deletion and update

### PR DESCRIPTION
**Pull request by atomrc**

### CODE REVIEW

File: app/script/storage/StorageSchemata.js [hunk 1]
```diff
@@ -310,6 +310,10 @@ z.storage.StorageSchemata = class StorageSchemata {
         },
         version: 15,
       },
+      {
+        schema: {},
```
Comment (atomrc) [inline app/script/storage/StorageSchemata.js]: We need an empty change in the DB cause dexie-observable is creating some extra tables and indexes

File: app/script/storage/StorageService.js [hunk 2]
```diff
@@ -72,42 +78,25 @@ class StorageService {
...
-            );
-        });
-      }
-    };
-    if (!this.db) {
-      // waiting for the DB to be initialized
-      return (this._afterDbInit = initHook);
-    }
-    initHook();
+    this.dbListeners.push({callback, store: storeName, type: 2});
```
Comment (bennycode) [inline app/script/storage/StorageService.js]: What's `type: 2`?

Comment (atomrc) [reply app/script/storage/StorageService.js]: This is the type of change we want to listen to. These are value sent by Dexie.
Do you think I should create a named variable to make that more explicit?

Comment (bennycode) [reply app/script/storage/StorageService.js]: That would be great.

**Pull request merged**